x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -205,7 +204,7 @@ protected LogicalPlan rule(Aggregate aggregate) {
                             var attr = aggFuncToAttr.get(af);
                             // the agg doesn't exist in the Aggregate, create an alias for it and save its attribute
                             if (attr == null) {
-                                var temporaryName = temporaryName(agg, af);
+                                var temporaryName = temporaryName(af, agg, counter[0]++);
```

Comment:
Changed the strategy to indicate the inner and outer expression as that can differ across rules - in some an aggregation is an inner expressions while in others is the outer one.